test/test_cuda.py

```diff
@@ -5429,6 +5429,78 @@ def test_resnet(self):
         self.checkFunction(m, [inp])
 
 
+    def test_allocate_in_thread_to_pool(self):
+
+        def foo():
+            return torch.rand([4], device="cuda")
+
+        pool = torch.cuda.graph_pool_handle()
+        graph, outputs = cudagraphify(foo, [], pool=pool)
+        device = outputs[0].device.index
+        del outputs
+
+        @contextlib.contextmanager
+        def _use_cuda_memory_pool_manager(device, existing_graph, mem_pool):
+            """
+            Context manager to use cuda graph pool for new allocations. If you use this manager
+            all cudagraph tensors in use should be reflected in the allocator or they will be overwritten.
+            existing_graph should already have been used in a capture, and the mem_pool must already exist.
+            """
+            capture_id = existing_graph.id()
+            torch._C._cuda_allocateThreadToPrivatePool(device, mem_pool)
+            torch._C._cuda_notifyCaptureBegin(device, capture_id, mem_pool)
+            try:
+                yield
+            finally:
+                torch._C._cuda_notifyCaptureAboutToEnd(device, capture_id)
```

Comment:
We need a capture id unique identifier that wont conflict with any other ongoing unique identifiers. The way to generate a unique identifier is through starting a cudagraph stream recording in cuda. 

Here I am just reusing a finished recording id with the graph, we could alternately start and end a cudastream recording, no strong opinions here..